### ISSUE TITLE
SimpleSeq2Seq

### ISSUE THREAD

**Issue by lfdharo** [farizrahman4u/seq2seq#108]
I have been trying to run the code in https://github.com/nicolas-ivanov/debug_seq2seq (commit 24888ce) but I have found one problem and a possible bug in SimpleSeq2Seq, may be you can help me to solve it.
- Bug: I found the following error when running the train.py script:
  Exception: Error when checking model target: expected recurrentcontainer_2 to have shape (None, 6, 512) but got array with shape (32, 6, 20001)
  Checking the code, and comparing the specifications of Simple2Seq2Seq and Seq2Seq, I think that the decoder must be defined in the following way (Following the same definition than in Seq2Seq):
  decoder.add(LSTMCell(hidden_dim, *_kwargs)) -> decoder.add(LSTMDecoderCell(output_dim=output_dim, hidden_dim=hidden_dim, batch_input_shape=(shape[0], output_dim), *_kwargs))

On the other hand, I have noticed that the sequence of layers is different for the encoder/decoder when comparing between Seq2Seq and SimpleSeq2Seq considering 2 LSTM layers, e.g.

[SimpleSeq2Seq]: encoder: LSTM + Dropout + LSTM    | decoder: Dropout + LSTM + Dropout + LSTM
[Seq2Seq]: LSTM + Dropout + LSTM + Dropout |  Dropout + LSTM + Dropout + LSTM

I think there could be an issue with Seq2Seq if the encoder ends in dropout and the decoder starts with dropout too.
- Problem: Replacing SimpleSeq2Seq for Seq2Seq solves the error mentioned above (also applying the proposed LSTMDecoderCell solution) but in both cases I just get loss=nan just from the first epoch and the outputs are always the same words, e.g.
  Epoch 1/1
  32/32 [==============================] - 4s - loss: nan - acc: 0.0781
  Epoch 1/1
  32/32 [==============================] - 4s - loss: nan - acc: 0.0677
  .....
  [what happened ?] -> [raining raining raining raining raining raining]
  [hello ?] -> [raining raining raining raining raining raining]

I'm using the following versions
Python = 2.7.6
Theano = 0.9.0dev3
Keras = 1.0.8
Recurrentshop = 0.0.1

I would appreciate any help on how to solve this issue. Thanks in advance.
